### PR TITLE
scrypt: add head reference and build dependencies

### PR DESCRIPTION
**Pull request by teslamint**

### CODE REVIEW

File: Library/Formula/scrypt.rb [hunk 1]
```diff
@@ -12,8 +13,13 @@ class Scrypt < Formula
   end
 
   depends_on "openssl"
+  if build.head?
```
Comment (mikemcquaid) [inline Library/Formula/scrypt.rb]: Put these in a `head do` block instead, thanks!

File: Library/Formula/scrypt.rb [hunk 2]
```diff
@@ -13,7 +14,13 @@ class Scrypt < Formula
 
   depends_on "openssl"
 
+  head do
```
Comment (DomT4) [inline Library/Formula/scrypt.rb]: Please include the `head` url in the `head` block, Thanks!

File: Library/Formula/scrypt.rb [hunk 3]
```diff
@@ -13,7 +14,13 @@ class Scrypt < Formula
...
+    depends_on "automake" => :build
+    depends_on "autoconf" => :build
+  end
+
   def install
+    system "autoreconf", "-i" if build.head?
```
Comment (DomT4) [inline Library/Formula/scrypt.rb]: Prefer `system "autoreconf", "-fvi" if build.head?` for consistency. Trying to unify these lines a little more.

**Comment by DomT4** [comment]
Merged in https://github.com/Homebrew/homebrew/commit/701c82f196e4a6f275ea473e82cb69040a3f941a. Thank you for your contribution to Homebrew @teslamint; we appreciate it! :smiley_cat:

**Pull request closed without merging**